### DISCUSSION TITLE
New TS011F Device from Aubess

### DISCUSSION THREAD

**Discussion by mendikant** [Koenkk/zigbee2mqtt#14899] [Devices]
Hello together!

I got a TS011F from Aubess today. I use zigbee2mqtt under iobroker. Have no idea about github and development unfortunately. 
Therefore my info here: The device was neither interviewed nor configured by the adapter. 

The shortest was mercator.js. I copied this to aubess.js and adapted it. After a restart the device was interviewed but not configured. But I can use it now. The current measurement works as well. Only problem with my file: The device has only one endpoint, not two. Unfortunately, adjusting the file did not work. I hope you get the rest right now? 

Thanks and greetings
Christian

============================

`const exposes = require('../lib/exposes');
const fz = require('../converters/fromZigbee');
const tz = require('../converters/toZigbee');
const reporting = require('../lib/reporting');
const e = exposes.presets;
const ea = exposes.access;

module.exports = [
    {
        fingerprint: [{modelID: 'TS011F', manufacturerName: '_TZ3000_hdopuwv6'}],
        model: 'TS011F',
        vendor: 'Aubess',
        description: 'Ikuü double outdoors power point',
        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report, fz.tuya_switch_power_outage_memory],
        toZigbee: [tz.on_off],
        exposes: [e.switch().withEndpoint('left'), e.switch().withEndpoint('right'),
            e.power().withEndpoint('left'), e.current().withEndpoint('left'),
            e.voltage().withEndpoint('left').withAccess(ea.STATE), e.energy()],
        endpoint: (device) => {
            return {left: 1, right: 2};
        },
        meta: {multiEndpoint: true},
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint1 = device.getEndpoint(1);
            const endpoint2 = device.getEndpoint(2);
            await reporting.bind(endpoint1, coordinatorEndpoint, ['genBasic', 'genOnOff', 'haElectricalMeasurement', 'seMetering']);
            await reporting.bind(endpoint2, coordinatorEndpoint, ['genOnOff']);
            await reporting.onOff(endpoint1);
            await reporting.onOff(endpoint1);
            await reporting.onOff(endpoint1);
            await reporting.rmsVoltage(endpoint1, {change: 5});
            await reporting.rmsCurrent(endpoint1, {change: 50});
            await reporting.activePower(endpoint1, {change: 1});
            await reporting.onOff(endpoint1);
            await reporting.onOff(endpoint2);
            endpoint1.saveClusterAttributeKeyValue('haElectricalMeasurement', {acCurrentDivisor: 1000, acCurrentMultiplier: 1});
            endpoint1.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
            device.save();
        },
    },
];`

**Comment by Koenkk** [comment]
`_TZ3000_hdopuwv6` is supported in z2m 1.28.2

**Reply by mendikant** [reply]
Thank you, i put the Entry into the tuya.ts and now it works...

**Reply by tumiro** [reply]
> `_TZ3000_hdopuwv6` is supported in z2m 1.28.2
Thank you Koenkk, so I can have hope that my Aubess power plug might one day start to cooperate. I still have:
"Zigbee2mqtt
Current version: 1.25.1-1"
Wonder why?

**Reply by loogfreak** [reply]
Hello, how can i update z2m in docker? My installation is on a Raspberry. I have the same struggle with die device from ausbess and my version is 1.26.0
